docs/design.md

```diff
@@ -138,7 +138,47 @@ upgrades, but must be configured at build time to choose one of these two
 strategies.
 
 In addition to the two image slots, the boot loader requires a scratch area to
-allow for reliable image swapping.
+allow for reliable image swapping. The scratch area must have a size that is
+enough to store at least the largest sector that is going to be swapped. Many
+devices have small equally sized flash sectors, eg 4K, while others have variable
+sized sectors where the largest sectors might be 128K or 256K, so the scratch
+must be big enough to store that. The scratch is only ever used when swapping
+firmware, which means only when doing an upgrade. Given that, there are two main
+reasons for using a larger size for the scratch. One is that the speed of an
+upgrade is proportional to the size, because larger size allows for less swap
+operations. Second is that flash wear will be more evenly distributed, because
+a single sector would be written twice the amount of times than using two
+sectors for example. To evaluate the ideal size of the scratch for your use
+case the following parameters are relevant:
+
+* the ratio of image size / scratch size
+* the number of erase cycles supported by the flash hardware
+
+The image size is used (instead of slot size) because only the slot's sectors
+that are actually used for storing the image are copied. The image/scratch ratio
+is the amount of times the scratch will be erased on every upgrade. The number
+of erase cycles divided by the image/scratch rate will give you the amount of
```

Comment:
s/rate/ratio/

Reply:
I'll pile on "number of times" here and on the previous line, instead of "amount of times" since there is already a typo to fix